src/platform/Linux/BLEManagerImpl.cpp

```diff
@@ -592,16 +592,16 @@ void BLEManagerImpl::DriveBLEState()
     // Initializes the Bluez BLE layer if needed.
     if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_Enabled && !mFlags.Has(Flags::kBluezBLELayerInitialized))
     {
-        err = InitBluezBleLayer(mIsCentral, nullptr, mBLEAdvConfig, mpEndpoint);
-        SuccessOrExit(err);
+        mEndpoint = Platform::MakeUnique<BluezEndpoint>(mIsCentral);
+        VerifyOrExit(mEndpoint, ChipLogError(DeviceLayer, "FAIL: memory allocation in %s", __func__));
```

Comment:
this does not set ERR to a value, so the exit will fail to run the "disabling" bit that it does for all other errors.

It seems like no memory should be handled like any other type of error ... should we set an err value here like no-mem as well?

Reply:
Thanks, fixed!